cmd/get/node_execution.go

```diff
@@ -52,151 +43,239 @@ const (
 	hyphenPrefix               = " - "
 )
 
-func NodeExecutionToProtoMessages(l []*admin.NodeExecution) []proto.Message {
-	messages := make([]proto.Message, 0, len(l))
-	for _, m := range l {
-		messages = append(messages, m)
+func getExecutionDetails(ctx context.Context, project, domain, execName, nodeName string, cmdCtx cmdCore.CommandContext) ([]*NodeExecutionWrapper, error) {
+	// Fetching Node execution details
+	nodeExecDetailsMap := map[string]*NodeExecutionWrapper{}
+	nExecDetails, err := getNodeExecDetailsInt(ctx, project, domain, execName, nodeName, "", nodeExecDetailsMap, cmdCtx)
+	if err != nil {
+		return nil, err
 	}
-	return messages
-}
 
-func NodeTaskExecutionToProtoMessages(l []*admin.TaskExecution) []proto.Message {
-	messages := make([]proto.Message, 0, len(l))
-	for _, m := range l {
-		messages = append(messages, m)
+	var nExecDetailsForView []*NodeExecutionWrapper
+	// Get the execution details only for the nodeId passed
+	if len(nodeName) > 0 {
+		// Fetch the last one which contains the nodeId details as previous ones are used to reach the nodeId
+		if nodeExecDetailsMap[nodeName] != nil {
+			nExecDetailsForView = append(nExecDetailsForView, nodeExecDetailsMap[nodeName])
+		}
+	} else {
+		nExecDetailsForView = nExecDetails
 	}
-	return messages
+	return nExecDetailsForView, nil
 }
 
-func getExecutionDetails(ctx context.Context, project, domain, name string, cmdCtx cmdCore.CommandContext) error {
-	adminPrinter := printer.Printer{}
+type StatusTrackerWrapper struct {
+	Phase string `json:"phase,omitempty"`
+	// Time at which the node execution began running.
+	StartedAt time.Time `json:"started_at,omitempty"`
+	// The amount of time the node execution spent running.
+	EndedAt time.Time `json:"ended_at,omitempty"`
+	// Time at which the node execution was created.
+	CreatedAt time.Time `json:"created_at,omitempty"`
+	// Time at which the node execution was last updated.
+	UpdatedAt time.Time `json:"updated_at,omitempty"`
+}
 
-	// Fetching Node execution details
-	nExecDetails, nodeExecToTaskExec, err := getNodeExecDetailsWithTasks(ctx, project, domain, name, cmdCtx)
-	if err != nil {
-		return err
-	}
+type NodeExecutionWrapper struct {
```

Comment:
Lets add comments for these structs, to explain - why do we need them, whats the goal

Reply:
```golang
type NodeExecutionClosure struct {
*NodeExecution
ChildNodes []*NodeExecutionWrapper `json:"child_nodes,omitempty"`
TaskExecutions []*TaskExecutionWrapper `json:"task_execs,omitempty"
}
```

Reply:
Hi @kumare3 , i will take these as a followup to this PR. facing issues with the marshalling of these Closure classes and needs to dug deeper.
For now we can go with these wrapper classes and i will soon send followup PR with this fix.

Reply:
Done